### PR TITLE
Remove workarounds

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Following up to https://github.com/aspnet/AspNetCore/pull/12557. I'll send a tell-mode email if this PR passes.

### CODE REVIEW

File: eng/Workarounds.targets [hunk 1]
```diff
@@ -54,21 +49,6 @@
...
-
-        This workaround was also applied to src\ProjectTemplates\test\Infrastructure\TemplateTests.props.in.
-      -->
-      <RuntimePackNamePatterns>Microsoft.NETCore.App.Runtime.**RID**</RuntimePackNamePatterns>
-      <AppHostPackNamePattern />
-      <AppHostRuntimeIdentifiers />
-    </KnownFrameworkReference>
-  </ItemGroup>
-
   <!-- Work around https://github.com/dotnet/cli/issues/11378. -->
```
Comment (dougbu) [inline eng/Workarounds.targets]: Remove this bit as well. This old workaround is not necessary now that we're using the preview 7 SDK. The last few comments about a new problem in the preview 8 SDK are worked around [above](https://github.com/aspnet/AspNetCore/pull/12812/files#diff-443ee38227003305d126960ccf383455R45-R50)

**Review by dougbu — CHANGES_REQUESTED** [review]
Please double-check if anything else is closed and available in our current dependencies. E.g. anything else fixed in the preview 7 SDK?

**Comment by JunTaoLuo** [comment]
I'll try addressing https://github.com/aspnet/AspNetCore/issues/11701 while we're removing workarounds.

**Comment by JunTaoLuo** [comment]
I've investigated this a little more and it seems like the workaround we want to remove for https://github.com/dotnet/cli/issues/11378 still affects building some projects for reference assemblies:
image

I'm not sure why this only seems to affect some (but not all) reference assembly projects but it seems like we can't remove the workaround just yet. This only seem to affect windows builds only and no other platforms. @dsplaisted would you know what to look for here? 

FYI, the binlogs that I'm looking at is here: https://dev.azure.com/dnceng/public/_build/results?buildId=292612&view=logs&jobId=d9e61dd5-50da-562f-684c-804952f2640b. 

For the time being, I'm adding back the workarounds.

**Comment by dougbu** [comment]
@dsplaisted as @JunTaoLuo described, we're still seeing
```
...FrameworkReferenceResolution.targets(274,5): error NETSDK1073: The FrameworkReference 'NETStandard.Library' was not recognized [...]
```
if we remove the dotnet/cli#11378 workaround (a conditional `<TransitiveFrameworkReference Remove="NETStandard.Library" />`).

With the 3.0.100-preview7-012821 .NET SDK, shouldn't we have picked up the bug fix?

Or, is the problem only fixed on CI agents with VS 16.2 or later (for example)? If this is relevant, the errors are visible with both BuildPool.Windows.10.Amd64.VS2019.BT.Open and BuildPool.Windows.10.Amd64.VS2019.Open agents. (Not positive what VS 2019 version those agents have installed at the moment.)

**Comment by dougbu** [comment]
@JunTaoLuo suggest getting this in and leaving #11701 as well as any improvements related to dotnet/cli#11378 for later. Your build was clean 😃

**Comment by JunTaoLuo** [comment]
Missing an approval but I can rage merge 😄

**Pull request merged**